src/process-events.c

```diff
@@ -125,7 +124,7 @@ struct bpf_map_def SEC("maps/tail_call_table") tail_call_table = {
         read_value(ptr, CRC_MM_STRUCT_EXE_FILE, &ptr, sizeof(ptr));                                     \
         read_value(ptr, CRC_FILE_F_INODE, &ptr, sizeof(ptr));                                           \
         read_value(ptr, CRC_INODE_I_SB, &sptr, sizeof(sptr));                                           \
-        read_value(sptr, CRC_SBLOCK_S_DEV, &i_rdev, sizeof(i_rdev));                                    \
+        read_value(sptr, CRC_SBLOCK_S_DEV, &i_dev, sizeof(i_dev));                                      \
```

Comment:
I see what was happening now - this is my fault. I was using the `i_rdev` value which is the device number of the file itself. `i_dev` is the device number of the device that the file is present on. So `i_rdev` will work for char/block devices (not relevant for us), whereas `i_dev` gives the information we're after.